packages/api/src/controllers/helpers.js

```diff
@@ -144,3 +145,27 @@ export async function sendgridEmail({
   SendgridMail.setApiKey(sendgridApiKey)
   await SendgridMail.send(msg)
 }
+
+export async function trackAction(userId, email, event, segmentApiKey) {
+  var analytics = new SegmentAnalytics(segmentApiKey, { flushAt: 1 })
+  // note: for development, add `{ flushAt: 1 }` as a second parameter to SegmentAnalytics
+  analytics.identify({
+    userId: userId,
+    traits: {
+      email: email,
+    },
+  })
+
+  const properties = {}
+  if ('properties' in event) {
+    for (const key in Object.keys(event.properties)) {
+      properties[key] = event.properties[key]
+    }
```

Comment:
Easier way to write this:
```suggestion
    properties = {...properties, ...event.properties}
```